lib/SILOptimizer/Mandatory/DiagnoseStaticExclusivity.cpp

```diff
@@ -562,18 +568,35 @@ static void diagnoseExclusivityViolation(const ConflictingAccess &Violation,
             CallsToSwap, Ctx, CallToReplace, Base, SwapIndex1, SwapIndex2);
     }
 
-    auto D =
-        diagnose(Ctx, MainAccess.getAccessLoc().getSourceLoc(),
-                 diag::exclusivity_access_required,
-                 PathDescription, AccessKindForMain, SuggestSwapAt);
-    D.highlight(RangeForMain);
-    if (SuggestSwapAt)
-      addSwapAtFixit(D, CallToReplace, Base, SwapIndex1, SwapIndex2,
-                     Ctx.SourceMgr);
+    if (isMoveOnly) {
+        auto D = diagnose(Ctx, MainAccess.getAccessLoc().getSourceLoc(),
+                          diag::exclusivity_access_required_moveonly,
```

Comment:
Duplicating this entire block seems unnecessary if the only difference is the diagnostic wording.

Reply:
I did this b/c the arguments to the diagnostics are different.